### PR TITLE
Generate CEN if not exists when sending report

### PR DESCRIPTION
**Pull request by i-schuetz**

### CODE REVIEW

File: CoEpi/repo/CoepiRepo.swift [hunk 1]
```diff
@@ -91,11 +91,15 @@ class CoEpiRepoImpl: CoEpiRepo {
...
     // TODO clarify with Rust lib, does it store the keys or we pass them
     func sendReport(report: CenReport) -> Completable {
-        if let lastCenKey = cenKeyDao.getLatestCENKey() { // TODO last n keys?
+        switch cenKeyDao.generateAndStoreCENKey() {  // TODO last n keys?
+        case .success(let key):
             // TODO clarify id
-            return api.postCenReport(cenReport: MyCenReport(id: "123", report: report, keys: lastCenKey.cenKey))
-        } else {
-            return Completable.error(RepoError.userHasNoCenKeys)
+            return api.postCenReport(cenReport: MyCenReport(id: "123", report: report, keys: key.cenKey))
```
Comment (duskoo) [inline CoEpi/repo/CoepiRepo.swift:97]: @i-schuetz We can generate random report id:
```
let id = UUID().uuidString
```

File: CoEpi/ble/ContactReceivedHandler.swift [hunk 2]
```diff
@@ -13,11 +14,20 @@ class ContactReceivedHandler {
...
+        switch cenKeyRepo.generateAndStoreCENKey() {
+        case .success(let key):
+            return cenLogic.generateCen(CENKey: key.cenKey)
+            //*** Scenario 1: https://docs.google.com/document/d/1f65V3PI214-uYfZLUZtm55kdVwoazIMqGJrxcYNI4eg/edit#
+            // iOS - Central + iOS - Peripheral -- so commenting out addNewContact
+            //addNewContactEvent(with: identifier)
+
+        case .failure(let error):
+            os_log("Couldn't generate CEN key: %@", type: .error, "\(error)")
+            // TODO clarify with BT lib how to handle error.
```
Comment (duskoo) [inline CoEpi/ble/ContactReceivedHandler.swift:26]: App running without CEN key does not make much sense.

File: CoEpi/repo/CoepiRepo.swift (hunk 1 above)
Comment (i-schuetz) [reply CoEpi/repo/CoepiRepo.swift:97]: We should ask the backend, not sure what they expect there.

File: CoEpi/ble/ContactReceivedHandler.swift (hunk 2 above)
Comment (i-schuetz) [reply CoEpi/ble/ContactReceivedHandler.swift:26]: Yep. We will address the edge cases / TODOs once the core functionality is finished.

**Pull request merged**